classes/task/delete_archived_object_metadata.php

```diff
@@ -0,0 +1,56 @@
+<?php
+// This file is part of Moodle - http://moodle.org/
+//
+// Moodle is free software: you can redistribute it and/or modify
+// it under the terms of the GNU General Public License as published by
+// the Free Software Foundation, either version 3 of the License, or
+// (at your option) any later version.
+//
+// Moodle is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+// GNU General Public License for more details.
+//
+// You should have received a copy of the GNU General Public License
+// along with Moodle.  If not, see <http://www.gnu.org/licenses/>.
+
+/**
+ * Task that checks for old archived objects, and removes their metadata
+ * (record) as it is no longer useful/relevant.
+ *
+ * @package   tool_objectfs
+ * @author    Kevin Pham <kevinpham@catalyst-au.net>
+ * @copyright Catalyst IT
+ * @license   http://www.gnu.org/copyleft/gpl.html GNU GPL v3 or later
+ */
+
+namespace tool_objectfs\task;
+
+defined('MOODLE_INTERNAL') || die();
+
+require_once(__DIR__ . '/../../lib.php');
+
+class delete_archived_object_metadata extends task {
+
+    /** @var string $stringname */
+    protected $stringname = 'delete_archived_object_metadata_task';
+
+    /**
+     * Execute task
+     */
+    public function execute() {
+        global $DB;
+
+        $wheresql = 'location = :location and timeduplicated < :ageforremoval';
+        $ageforremoval = 30 * DAYSECS;
```

Comment:
turn this into a setting which defaults to 0 which means don't ever delete

Reply:
![image](https://user-images.githubusercontent.com/9924643/144541587-227309d4-0bc2-4270-b827-ec551bf7c18a.png)

I've added this under general settings, please let me know if the wording needs to be updated.